File: codegen/codegen-elixir/src/test/resources/ir/rest_xml_decode_get_name_response.expected.ex
@spec decode_get_name_response(map()) ::
        {:ok, HttpServiceTypes.GetNameOutput.t()} | {:error, term()}
def decode_get_name_response(%{:status => 200, :headers => headers, :body => body}) do
  parsed = case parse_xml_root(body, "GetNameOutput") do
    {:ok, root} -> root
    {:error, _} -> :nil
  end
  name = case parsed do
    nil -> :nil
    _ -> xml_child_text(parsed, "Name")
  end
  {:ok, %Types.GetNameOutput{name: name}}
end
def decode_get_name_response(%{:status => status, :body => body}) do
  {:error, {:unknown_error, status, body}}
end

File: codegen/codegen-erlang/src/test/resources/dsl/rest_xml_decode_get_name_response.expected.erl
%% @doc Decode REST-XML response for smithy.beam.demo.http#GetName.
-spec decode_get_name_response(#http_response{}) -> {'ok', get_name_output()} | {'error', term()}.
decode_get_name_response(#http_response{status = 200, headers = Headers, body = Body}) ->
    Parsed =
        case parse_xml_root(Body, <<"GetNameOutput">>) of
            {ok, Root} -> Root;
            {error, _} -> undefined
        end,
    Name =
        case Parsed of
            undefined -> undefined;
            _ -> xml_child_text(Parsed, <<"Name">>)
        end,
    {ok, #get_name_output{
        name = Name
    }};
decode_get_name_response(#http_response{status = Status, body = Body}) ->
    decode_rest_xml_error(Status, Body).
